### PR TITLE
Allow to pass extra env vars to the nodes

### PR DESCRIPTION
**Pull request by szczeles**

### CODE REVIEW

File: kedro_azureml/cli_functions.py [hunk 1]
```diff
@@ -129,3 +132,11 @@ def verify_configuration_directory_for_azure(click_context, ctx: CliContext):
         )
         if not click.confirm(click.style(msg, fg="yellow")):
             click_context.exit(2)
+
+
+def parse_extra_env_params(extra_env):
+    for entry in extra_env:
+        if not re.match(".+=.*", entry):
```
Comment (marrrcin) [inline kedro_azureml/cli_functions.py:139]: Maybe we should be more strict on the left side, e.g. `[A-Za-z_]+=.*`?

File: kedro_azureml/cli_functions.py [hunk 2]
```diff
@@ -129,3 +132,11 @@ def verify_configuration_directory_for_azure(click_context, ctx: CliContext):
...
+            raise Exception(f"Invalid env-var: {entry}, expected format: KEY=VALUE")
+
+    return {entry.split("=")[0]: entry.split("=")[1] for entry in extra_env}
```
Comment (marrrcin) [inline kedro_azureml/cli_functions.py:142]:
```
{(e := entry.split("="))[0]: e[1] for entry in extra_env}
```
?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 ## [Unreleased]
 
+-   Ability to pass extra environment variables to the Kedro nodes using `--env-var` option
```
Comment (marrrcin) [inline CHANGELOG.md:5]: Let's also add `--env-var GETINDATA=ROCKS!` to e2e tests to cover this feature.

File: kedro_azureml/cli_functions.py (hunk 1 above)
Comment (szczeles) [reply kedro_azureml/cli_functions.py:139]: Okay! I will allow digits too

File: kedro_azureml/cli_functions.py (hunk 2 above)
Comment (szczeles) [reply kedro_azureml/cli_functions.py:142]: Walrus FTW!

File: CHANGELOG.md (hunk 3 above)
Comment (szczeles) [reply CHANGELOG.md:5]: :sunglasses:

**Comment by szczeles** [comment]
End-to-end tests confim env var being populated:
image

**Pull request merged**